dronekit_sitl/__init__.py

```diff
@@ -137,14 +138,59 @@ def check_complete(count, block_size, total_size):
     if verbose:
         print('Ready to boot.')
 
+sitl_instance_count = 0
+
+class ArdupilotCapabilities():
+    def __init__(self, path):
+        # Load the binary for primitive feature detection.
+        elf = open(path, 'rb').read()
+
+        # pysim is required for earlier SITL builds
+        # lacking --home or --model params.
+        need_sim = not b'--home' in elf or not b'--model' in elf
+        self.using_sim = need_sim
+
+        if b'ardupilot/APMrover2' in elf:
+            self.model = 'rover'
+        elif b'ardupilot/ArduPlane' in elf:
+            self.model = 'plane'
+        else:
+            self.model = 'quad'
+
+
+        try:
+            helptext = subprocess.Popen([path, '--help'], stdout=subprocess.PIPE).communicate()[0]
+        except OSError as e:
+            print("Failed to run (%s)" % (path,))
+
+        self.has_defaults_path = "--defaults path" in helptext
+
+
+def main_thread():
+    for thread in threading.enumerate():
+        if thread.name == "MainThread":
+            return thread
+    return None # should not get to here!  Raise an exception instead?
```

Comment:
this should be a candidate for a test case to force this condition, I think we should default to an exception, given this lib is sometimes used for testing other systems it would be a failure on our part to silently fail